### PR TITLE
delete default css

### PR DESCRIPTION
**Pull request by ralfiannor**
Implement #253

### CODE REVIEW

**Comment by matthewmueller** [comment]
Thanks for the PR @ralfiannor and for kicking this off! I don't want to remove the default.css without providing an alternative solution. I've wrote my thoughts up in #253. Let me know if you want to drive one of those solutions forward or if you have any other ideas!

File: framework/view/ssr/svelte.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ const defaultLayout = {
         <html>
           <head>
             <meta charset="utf-8"/>
-            <style>${defaultCSS}</style>
+			<link rel="stylesheet" href="default.css">
```
Comment (matthewmueller) [inline framework/view/ssr/svelte.ts:61]: Formatting

File: framework/view/ssr/svelte.js [hunk 2]
```diff
@@ -327,7 +327,7 @@ var defaultLayout = {
         <html>
           <head>
             <meta charset="utf-8"/>
-            <style>${defaultCSS}</style>
+      			<link rel="stylesheet" href="default.css">
```
Comment (matthewmueller) [inline framework/view/ssr/svelte.js:330]: Spacing looks off

File: internal/cli/create/create_test.go [hunk 3]
```diff
@@ -38,6 +38,7 @@ func TestCreateOutsideGoPath(t *testing.T) {
 	is.NoErr(td.Exists("go.sum"))
 	is.NoErr(td.Exists("package.json"))
 	is.NoErr(td.Exists("package-lock.json"))
+	is.NoErr(td.Exists("public/default.css"))
```
Comment (matthewmueller) [inline internal/cli/create/create_test.go:41]: I think we're missing some test cases still:
1. The scaffolded `public/default.css` gets deleted but views and public files still work as expected.
2. Someone brings their own `public/default.css` and that gets picked up and used.

File: internal/embedded/embedded.go [hunk 4]
```diff
@@ -7,7 +7,19 @@ import (
...
+//go:embed normalize.css
+var normalize string
+
 // Favicon returns the favicon data
 func Favicon() []byte {
 	return favicon
 }
+
+// DefaultCss returns the default css data
+func DefaultCss(css string) string {
```
Comment (matthewmueller) [inline internal/embedded/embedded.go:19]: I think we can just use function names like `func Normalize() []byte` or `func Reset() []byte` since it's pretty clear where the context comes from.

**Review by matthewmueller — COMMENTED** [review]
Thanks for working on this! I'd just like us to clean a few things up and add a few more tests (maybe in framework/public/public_test.go) and then I think we're good to go!

File: internal/cli/create/create_test.go (hunk 3 above)
Comment (ralfiannor) [reply internal/cli/create/create_test.go:42]: Added the test case in `public_test.go`

File: framework/controller/controller_test.go [hunk 5]
```diff
@@ -1285,7 +1285,6 @@ func TestViewUnnamed(t *testing.T) {
 	is.NoErr(err)
 	is.NoErr(res.DiffHeaders(`
 		HTTP/1.1 200 OK
-		Transfer-Encoding: chunked
```
Comment (matthewmueller) [inline framework/controller/controller_test.go:1288]: I'm quite surprised the changes affected these tests. Do you know why you needed to remove this?

File: internal/embedded/embedded.go [hunk 6]
```diff
@@ -7,7 +7,20 @@ import (
...
+
+// NormalizeCss returns the normalize css data
+func NormalizeCss() []byte {
+	return normalize
+}
+
+// EmptyCss reset the default css data
+func EmptyCss() []byte {
+	return []byte("/* No Default CSS Loaded */")
+}
```
Comment (matthewmueller) [inline internal/embedded/embedded.go:26]: It'd be better if the view didn't include the `<link />` if a `public/default.css` isn't present since that would avoid a roundtrip for this CSS sheet. 

Then we can also remove this function.

File: internal/cli/cli.go [hunk 7]
```diff
@@ -49,6 +49,7 @@ func (c *CLI) Run(ctx context.Context, args ...string) error {
 		cmd := create.New(cmd, c.in)
 		cli := cli.Command("create", "create a new app")
 		cli.Arg("dir").String(&cmd.Dir)
+		cli.Flag("css", "add a css").String(&cmd.Css).Default("normalize")
```
Comment (matthewmueller) [inline internal/cli/cli.go:52]: Let's remove this flag for now, I'm not ready to expose it just yet, especially since it's not really possible to select an alternative.

File: framework/public/public_test.go [hunk 8]
```diff
@@ -144,8 +149,20 @@ func TestGetChangeGet(t *testing.T) {
...
+	is.Equal(res.Body().Bytes(), defaultCss)
+
+	// default.css changed
+	defaultCss2 := []byte("/* changed default.css */")
+	td.BFiles["public/default.css"] = defaultCss2
+	is.NoErr(td.Write(ctx))
+	res, err = app.Get("/default.css")
+	is.NoErr(err)
+	is.Equal(200, res.Status())
+	is.Equal(res.Body().Bytes(), defaultCss2)
```
Comment (matthewmueller) [inline framework/public/public_test.go:165]: It's a bit hard to follow these changes because you've added new conditions to existing tests. 

Let's try to only touch existing tests that need to be updated for the new behavior, then add new tests for the new behavior.

File: framework/view/ssr/svelte.js [hunk 9]
```diff
@@ -327,7 +327,7 @@ var defaultLayout = {
         <html>
           <head>
             <meta charset="utf-8"/>
-            <style>${defaultCSS}</style>
+            <link rel="stylesheet" href="default.css" />
```
Comment (matthewmueller) [inline framework/view/ssr/svelte.js:330]: Do you mind trying to feed the configuration through so you can remove this if `public/default.css` doesn't exist? Let me know if it's difficult and I'll give it a go in this PR.

**Review by matthewmueller — CHANGES_REQUESTED** [review]
Just reviewed. This is getting closer @ralfiannor!

**Comment by matthewmueller** [comment]
I just realized we should have the same treatment for the public/favicon.ico: scaffold a default one, but then give users the option to override it or delete it.

File: framework/view/ssr/svelte.js (hunk 9 above)
Comment (ralfiannor) [reply framework/view/ssr/svelte.js:330]: Will try. I will add the checking file existing in `ssr.go` but will call our logic each page loaded. Is it okay ?

Comment (matthewmueller) [reply framework/view/ssr/svelte.js:330]: I think it should be done at compile time actually

File: framework/controller/controller_test.go (hunk 5 above)
Comment (ralfiannor) [reply framework/controller/controller_test.go:1288]: Reverted. But I still get an error on it while running the unit test. I use `go 1.7` and `node 18.9`

**Comment by ralfiannor** [comment]
> I just realized we should also do the same thing for `public/favicon.ico`: scaffold a default one, but then give users the option to override it or delete it.

Can we do it on another PR?

File: framework/public/public_test.go (hunk 8 above)
Comment (ralfiannor) [reply framework/public/public_test.go]: Reverted. No need to update on `public.go` just update on `ssr.go`

File: internal/embedded/embedded.go (hunk 6 above)
Comment (ralfiannor) [reply internal/embedded/embedded.go]: Done

File: internal/cli/cli.go (hunk 7 above)
Comment (ralfiannor) [reply internal/cli/cli.go]: Done

**Review by matthewmueller — APPROVED** [review]

**Comment by matthewmueller** [comment]
Thanks for continuing to push this forward @ralfiannor! 

Do you know why Go is adding/removing `Transfer-Encoding: chunked`? 

Let's just remove the `Transfer-Encoding: chunked` in the view tests and I think this is good to go!

**Comment by ralfiannor** [comment]
> Do you happen to know the condition in which Go is adding/removing `Transfer-Encoding: chunked`?

Yes, it happen when we added <style></style> the chunked also added

**Comment by matthewmueller** [comment]
Thanks @ralfiannor! I just pulled it down and did some manual testing. Two issues popped up:

1. Welcome page no longer shows up when creating a new project
2. Renaming or removing the `default.css` requires you to restart the `bud` process

Adjusting the global `svelteRuntime` is a bit hacky. If you're up for cleaning this up, a better solution would be to pass this information through the `svelte.gotext` template and then reference that information in the `svelte.ts` file. I'm okay with cleaning this up later if you're not sure how to do this.

Here's a diff with the fixes:

```diff
diff --git a/framework/app/app_test.go b/framework/app/app_test.go
index b5ba644..500a117 100644
--- a/framework/app/app_test.go
+++ b/framework/app/app_test.go
@@ -30,3 +30,26 @@ func TestWelcome(t *testing.T) {
 	is.NoErr(td.Exists("bud/app"))
 	is.NoErr(app.Close())
 }
+
+func TestWelcomeWithPublic(t *testing.T) {
+	is := is.New(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	dir := t.TempDir()
+	td := testdir.New(dir)
+	td.Files["public/default.css"] = `* { box-sizing: border-box; }`
+	is.NoErr(td.Write(ctx))
+	cli := testcli.New(dir)
+	is.NoErr(td.NotExists("bud/app"))
+	app, err := cli.Start(ctx, "run")
+	is.NoErr(err)
+	res, err := app.Get("/")
+	is.NoErr(err)
+	is.Equal(res.Status(), 200)
+	is.In(res.Body().String(), "Hey Bud")
+	is.In(res.Body().String(), "Hey Bud") // should work multiple times
+	is.Equal(app.Stdout(), "")
+	is.Equal(app.Stderr(), "")
+	is.NoErr(td.Exists("bud/app"))
+	is.NoErr(app.Close())
+}
diff --git a/framework/view/ssr/ssr.go b/framework/view/ssr/ssr.go
index 7116bcd..8bab21c 100644
--- a/framework/view/ssr/ssr.go
+++ b/framework/view/ssr/ssr.go
@@ -50,9 +50,10 @@ type Compiler struct {
 
 func (c *Compiler) Compile(ctx context.Context, fsys budfs.FS) ([]byte, error) {
 	dir := c.module.Directory()
-
 	if existCss := vfs.Exist(fsys, "public/default.css"); nil == existCss {
 		svelteRuntime = strings.Replace(svelteRuntime, `<!-- default css -->`, `<link rel="stylesheet" href="default.css">`, 1)
+	} else {
+		svelteRuntime = strings.Replace(svelteRuntime, `<link rel="stylesheet" href="default.css">`, `<!-- default css -->`, 1)
 	}
 
 	result := esbuild.Build(esbuild.BuildOptions{
diff --git a/framework/web/loader.go b/framework/web/loader.go
index 5f0d149..e4695b8 100644
--- a/framework/web/loader.go
+++ b/framework/web/loader.go
@@ -52,7 +52,10 @@ func (l *loader) Load() (state *State, err error) {
 	l.imports.AddNamed("webrt", "github.com/livebud/bud/framework/web/webrt")
 	l.imports.AddNamed("router", "github.com/livebud/bud/package/router")
 	// Show the welcome page if we don't have controllers, views or public files
-	if len(exist) == 0 {
+	//
+	// TODO: welcome should be part of the default controller, then we can support
+	// a simple static file server without generating the welcome page.
+	if len(exist) == 0 || (len(exist) == 1 && exist["bud/internal/app/public/public.go"]) {
 		l.imports.AddNamed("welcome", "github.com/livebud/bud/framework/web/welcome")
 		state.ShowWelcome = true
 		state.Imports = l.imports.List()

```

**Comment by matthewmueller** [comment]
Any chance of wrapping this up @ralfiannor ?
